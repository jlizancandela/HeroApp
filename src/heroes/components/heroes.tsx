import { getHeroes } from '../helper/getheroes';
import { HeroCard } from './hero';

export const Heroes = ({ publisher }: { publisher: string }) => {
  const heroes = getHeroes(publisher);

  return (
    <div className="d-flex flex-row flex-wrap gap-3 mb-5">
      {heroes.map((hero) => (
        <HeroCard key={hero.id} hero={hero} />
      ))}
    </div>
  );
};
