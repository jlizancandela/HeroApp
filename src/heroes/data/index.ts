import { Hero } from "./types";
import data from "./heroes.json";
export const heroes: Hero[] = data;
export * from "./types";
