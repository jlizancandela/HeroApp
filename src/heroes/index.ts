export { MarvelPage } from './pages/marvel';
export { DcPage } from './pages/dc';
export { HeroPage } from './pages/hero';
export { SearchPage } from './pages/search';
