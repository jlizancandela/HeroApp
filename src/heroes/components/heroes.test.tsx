import { render, screen } from '@testing-library/react';
import { Heroes } from './heroes';

describe('Heroes', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<Heroes publisher="marvel" />);
    expect(baseElement).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
    expect(screen.getByText('Spider Man')).toBeInTheDocument();
  });

  it('should render successfully', () => {
    const { baseElement } = render(<Heroes publisher="dc" />);
    expect(baseElement).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
    expect(screen.getByText('Batman')).toBeInTheDocument();
  });
});
