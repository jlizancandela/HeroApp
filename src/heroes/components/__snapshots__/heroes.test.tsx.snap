// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Heroes > should render successfully 1`] = `
<body
  data-aos-delay="0"
  data-aos-duration="400"
  data-aos-easing="ease"
>
  <div>
    <div
      class="d-flex flex-row flex-wrap gap-3 mb-5"
    >
      <div
        class="card aos-init aos-animate"
        data-aos="fade-up"
        data-aos-duration="2000"
        style="width: 18rem;"
      >
        <img
          alt="marvel-spider"
          class="card-img-top"
          src="/heroes/marvel-spider.jpg"
        />
        <div
          class="card-body d-flex flex-column flex-fill"
        >
          <h5
            class="card-title"
          >
            Spider Man
          </h5>
          <p
            class="card-text"
          >
            Peter Parker
          </p>
          <p
            class="card-text"
          />
          <a
            class="btn btn-dark mt-auto"
            href="/hero/marvel-spider"
          >
            Mas información
          </a>
        </div>
      </div>
      <div
        class="card aos-init aos-animate"
        data-aos="fade-up"
        data-aos-duration="2000"
        style="width: 18rem;"
      >
        <img
          alt="marvel-captain"
          class="card-img-top"
          src="/heroes/marvel-captain.jpg"
        />
        <div
          class="card-body d-flex flex-column flex-fill"
        >
          <h5
            class="card-title"
          >
            Captain America
          </h5>
          <p
            class="card-text"
          >
            Steve Rogers
          </p>
          <p
            class="card-text"
          />
          <a
            class="btn btn-dark mt-auto"
            href="/hero/marvel-captain"
          >
            Mas información
          </a>
        </div>
      </div>
      <div
        class="card aos-init aos-animate"
        data-aos="fade-up"
        data-aos-duration="2000"
        style="width: 18rem;"
      >
        <img
          alt="marvel-iron"
          class="card-img-top"
          src="/heroes/marvel-iron.jpg"
        />
        <div
          class="card-body d-flex flex-column flex-fill"
        >
          <h5
            class="card-title"
          >
            Iron Man
          </h5>
          <p
            class="card-text"
          >
            Tony Stark
          </p>
          <p
            class="card-text"
          />
          <a
            class="btn btn-dark mt-auto"
            href="/hero/marvel-iron"
          >
            Mas información
          </a>
        </div>
      </div>
      <div
        class="card aos-init aos-animate"
        data-aos="fade-up"
        data-aos-duration="2000"
        style="width: 18rem;"
      >
        <img
          alt="marvel-thor"
          class="card-img-top"
          src="/heroes/marvel-thor.jpg"
        />
        <div
          class="card-body d-flex flex-column flex-fill"
        >
          <h5
            class="card-title"
          >
            Thor
          </h5>
          <p
            class="card-text"
          >
            Thor Odinson
          </p>
          <p
            class="card-text"
          />
          <a
            class="btn btn-dark mt-auto"
            href="/hero/marvel-thor"
          >
            Mas información
          </a>
        </div>
      </div>
      <div
        class="card aos-init aos-animate"
        data-aos="fade-up"
        data-aos-duration="2000"
        style="width: 18rem;"
      >
        <img
          alt="marvel-hulk"
          class="card-img-top"
          src="/heroes/marvel-hulk.jpg"
        />
        <div
          class="card-body d-flex flex-column flex-fill"
        >
          <h5
            class="card-title"
          >
            Hulk
          </h5>
          <p
            class="card-text"
          >
            Bruce Banner
          </p>
          <p
            class="card-text"
          />
          <a
            class="btn btn-dark mt-auto"
            href="/hero/marvel-hulk"
          >
            Mas información
          </a>
        </div>
      </div>
      <div
        class="card aos-init aos-animate"
        data-aos="fade-up"
        data-aos-duration="2000"
        style="width: 18rem;"
      >
        <img
          alt="marvel-wolverine"
          class="card-img-top"
          src="/heroes/marvel-wolverine.jpg"
        />
        <div
          class="card-body d-flex flex-column flex-fill"
        >
          <h5
            class="card-title"
          >
            Wolverine
          </h5>
          <p
            class="card-text"
          >
            James Howlett
          </p>
          <p
            class="card-text"
          />
          <a
            class="btn btn-dark mt-auto"
            href="/hero/marvel-wolverine"
          >
            Mas información
          </a>
        </div>
      </div>
      <div
        class="card aos-init aos-animate"
        data-aos="fade-up"
        data-aos-duration="2000"
        style="width: 18rem;"
      >
        <img
          alt="marvel-daredevil"
          class="card-img-top"
          src="/heroes/marvel-daredevil.jpg"
        />
        <div
          class="card-body d-flex flex-column flex-fill"
        >
          <h5
            class="card-title"
          >
            Daredevil
          </h5>
          <p
            class="card-text"
          >
            Matthew Michael Murdock
          </p>
          <p
            class="card-text"
          />
          <a
            class="btn btn-dark mt-auto"
            href="/hero/marvel-daredevil"
          >
            Mas información
          </a>
        </div>
      </div>
      <div
        class="card aos-init aos-animate"
        data-aos="fade-up"
        data-aos-duration="2000"
        style="width: 18rem;"
      >
        <img
          alt="marvel-hawkeye"
          class="card-img-top"
          src="/heroes/marvel-hawkeye.jpg"
        />
        <div
          class="card-body d-flex flex-column flex-fill"
        >
          <h5
            class="card-title"
          >
            Hawkeye
          </h5>
          <p
            class="card-text"
          >
            Clinton Francis Barton
          </p>
          <p
            class="card-text"
          />
          <a
            class="btn btn-dark mt-auto"
            href="/hero/marvel-hawkeye"
          >
            Mas información
          </a>
        </div>
      </div>
      <div
        class="card aos-init aos-animate"
        data-aos="fade-up"
        data-aos-duration="2000"
        style="width: 18rem;"
      >
        <img
          alt="marvel-cyclops"
          class="card-img-top"
          src="/heroes/marvel-cyclops.jpg"
        />
        <div
          class="card-body d-flex flex-column flex-fill"
        >
          <h5
            class="card-title"
          >
            Cyclops
          </h5>
          <p
            class="card-text"
          >
            Scott Summers
          </p>
          <p
            class="card-text"
          />
          <a
            class="btn btn-dark mt-auto"
            href="/hero/marvel-cyclops"
          >
            Mas información
          </a>
        </div>
      </div>
      <div
        class="card aos-init aos-animate"
        data-aos="fade-up"
        data-aos-duration="2000"
        style="width: 18rem;"
      >
        <img
          alt="marvel-silver"
          class="card-img-top"
          src="/heroes/marvel-silver.jpg"
        />
        <div
          class="card-body d-flex flex-column flex-fill"
        >
          <h5
            class="card-title"
          >
            Silver Surfer
          </h5>
          <p
            class="card-text"
          >
            Norrin Radd
          </p>
          <p
            class="card-text"
          />
          <a
            class="btn btn-dark mt-auto"
            href="/hero/marvel-silver"
          >
            Mas información
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Heroes > should render successfully 2`] = `
<body
  data-aos-delay="0"
  data-aos-duration="400"
  data-aos-easing="ease"
>
  <div>
    <div
      class="d-flex flex-row flex-wrap gap-3 mb-5"
    >
      <div
        class="card aos-init aos-animate"
        data-aos="fade-up"
        data-aos-duration="2000"
        style="width: 18rem;"
      >
        <img
          alt="dc-batman"
          class="card-img-top"
          src="/heroes/dc-batman.jpg"
        />
        <div
          class="card-body d-flex flex-column flex-fill"
        >
          <h5
            class="card-title"
          >
            Batman
          </h5>
          <p
            class="card-text"
          >
            Bruce Wayne
          </p>
          <p
            class="card-text"
          />
          <a
            class="btn btn-dark mt-auto"
            href="/hero/dc-batman"
          >
            Mas información
          </a>
        </div>
      </div>
      <div
        class="card aos-init aos-animate"
        data-aos="fade-up"
        data-aos-duration="2000"
        style="width: 18rem;"
      >
        <img
          alt="dc-superman"
          class="card-img-top"
          src="/heroes/dc-superman.jpg"
        />
        <div
          class="card-body d-flex flex-column flex-fill"
        >
          <h5
            class="card-title"
          >
            Superman
          </h5>
          <p
            class="card-text"
          >
            Kal-El
          </p>
          <p
            class="card-text"
          />
          <a
            class="btn btn-dark mt-auto"
            href="/hero/dc-superman"
          >
            Mas información
          </a>
        </div>
      </div>
      <div
        class="card aos-init aos-animate"
        data-aos="fade-up"
        data-aos-duration="2000"
        style="width: 18rem;"
      >
        <img
          alt="dc-flash"
          class="card-img-top"
          src="/heroes/dc-flash.jpg"
        />
        <div
          class="card-body d-flex flex-column flex-fill"
        >
          <h5
            class="card-title"
          >
            Flash
          </h5>
          <p
            class="card-text"
          >
            Jay Garrick
          </p>
          <p
            class="card-text"
          >
            Jay Garrick, Barry Allen, Wally West, Bart Allen
          </p>
          <a
            class="btn btn-dark mt-auto"
            href="/hero/dc-flash"
          >
            Mas información
          </a>
        </div>
      </div>
      <div
        class="card aos-init aos-animate"
        data-aos="fade-up"
        data-aos-duration="2000"
        style="width: 18rem;"
      >
        <img
          alt="dc-green"
          class="card-img-top"
          src="/heroes/dc-green.jpg"
        />
        <div
          class="card-body d-flex flex-column flex-fill"
        >
          <h5
            class="card-title"
          >
            Green Lantern
          </h5>
          <p
            class="card-text"
          >
            Alan Scott
          </p>
          <p
            class="card-text"
          >
            Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz
          </p>
          <a
            class="btn btn-dark mt-auto"
            href="/hero/dc-green"
          >
            Mas información
          </a>
        </div>
      </div>
      <div
        class="card aos-init aos-animate"
        data-aos="fade-up"
        data-aos-duration="2000"
        style="width: 18rem;"
      >
        <img
          alt="dc-arrow"
          class="card-img-top"
          src="/heroes/dc-arrow.jpg"
        />
        <div
          class="card-body d-flex flex-column flex-fill"
        >
          <h5
            class="card-title"
          >
            Green Arrow
          </h5>
          <p
            class="card-text"
          >
            Oliver Queen
          </p>
          <p
            class="card-text"
          />
          <a
            class="btn btn-dark mt-auto"
            href="/hero/dc-arrow"
          >
            Mas información
          </a>
        </div>
      </div>
      <div
        class="card aos-init aos-animate"
        data-aos="fade-up"
        data-aos-duration="2000"
        style="width: 18rem;"
      >
        <img
          alt="dc-wonder"
          class="card-img-top"
          src="/heroes/dc-wonder.jpg"
        />
        <div
          class="card-body d-flex flex-column flex-fill"
        >
          <h5
            class="card-title"
          >
            Wonder Woman
          </h5>
          <p
            class="card-text"
          >
            Princess Diana
          </p>
          <p
            class="card-text"
          />
          <a
            class="btn btn-dark mt-auto"
            href="/hero/dc-wonder"
          >
            Mas información
          </a>
        </div>
      </div>
      <div
        class="card aos-init aos-animate"
        data-aos="fade-up"
        data-aos-duration="2000"
        style="width: 18rem;"
      >
        <img
          alt="dc-martian"
          class="card-img-top"
          src="/heroes/dc-martian.jpg"
        />
        <div
          class="card-body d-flex flex-column flex-fill"
        >
          <h5
            class="card-title"
          >
            Martian Manhunter
          </h5>
          <p
            class="card-text"
          >
            J'onn J'onzz
          </p>
          <p
            class="card-text"
          >
            Martian Manhunter
          </p>
          <a
            class="btn btn-dark mt-auto"
            href="/hero/dc-martian"
          >
            Mas información
          </a>
        </div>
      </div>
      <div
        class="card aos-init aos-animate"
        data-aos="fade-up"
        data-aos-duration="2000"
        style="width: 18rem;"
      >
        <img
          alt="dc-robin"
          class="card-img-top"
          src="/heroes/dc-robin.jpg"
        />
        <div
          class="card-body d-flex flex-column flex-fill"
        >
          <h5
            class="card-title"
          >
            Robin/Nightwing
          </h5>
          <p
            class="card-text"
          >
            Dick Grayson
          </p>
          <p
            class="card-text"
          />
          <a
            class="btn btn-dark mt-auto"
            href="/hero/dc-robin"
          >
            Mas información
          </a>
        </div>
      </div>
      <div
        class="card aos-init aos-animate"
        data-aos="fade-up"
        data-aos-duration="2000"
        style="width: 18rem;"
      >
        <img
          alt="dc-blue"
          class="card-img-top"
          src="/heroes/dc-blue.jpg"
        />
        <div
          class="card-body d-flex flex-column flex-fill"
        >
          <h5
            class="card-title"
          >
            Blue Beetle
          </h5>
          <p
            class="card-text"
          >
            Dan Garret
          </p>
          <p
            class="card-text"
          >
            Dan Garret, Ted Kord, Jaime Reyes
          </p>
          <a
            class="btn btn-dark mt-auto"
            href="/hero/dc-blue"
          >
            Mas información
          </a>
        </div>
      </div>
      <div
        class="card aos-init aos-animate"
        data-aos="fade-up"
        data-aos-duration="2000"
        style="width: 18rem;"
      >
        <img
          alt="dc-black"
          class="card-img-top"
          src="/heroes/dc-black.jpg"
        />
        <div
          class="card-body d-flex flex-column flex-fill"
        >
          <h5
            class="card-title"
          >
            Black Canary
          </h5>
          <p
            class="card-text"
          >
            Dinah Drake
          </p>
          <p
            class="card-text"
          >
            Dinah Drake, Dinah Lance
          </p>
          <a
            class="btn btn-dark mt-auto"
            href="/hero/dc-black"
          >
            Mas información
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;
